_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw_vaapi.cpp

```diff
@@ -332,7 +332,8 @@ mfxStatus VideoDECODEVP8_HW::PackHeaders(mfxBitstream *p_bistream)
      uint8_t *bistreamData = (uint8_t *)m_p_video_accelerator->GetCompBuffer(VASliceDataBufferType, &compBufBs, p_bistream->DataLength - offset);
 
      uint8_t *pBuffer = (uint8_t*) p_bistream->Data;
-
+     if(size > compBufBs->GetBufferSize())
```

Comment:
Hmm. On Linux we control size of VASliceDataBufferType requested from driver. So instead of returning the error we just need to allocate a proper size VASliceDataBufferType buffer in advanced.

Reply:
Thanks for your comment, code updated.